test/test_classifier.rb

```diff
@@ -50,7 +50,7 @@ def test_classify_ambiguous_languages
       next unless languages.length > 1
 
       results = Classifier.classify(Samples.cache, File.read(sample[:path]), languages)
-      assert_equal language.name, results.first[0], "#{sample[:path]}\n#{results.inspect}"
+      assert results.map{|key,value| key}.include?(language.name), "#{sample[:path]}\n#{results.inspect}"
```

Comment:
Can you tell exactly why YAML is first in the list when:

* JSON is passed `samples/JSON/filenames/.eslintrc`
* heuristic is correctly classifying ` samples/JSON/filenames/.eslintrc` as JSON and ` samples/YAML/filenames/.eslintrc` as YAML.